### PR TITLE
[master] Porting #49837 to master

### PR DESCRIPTION
**Pull request by garethgreenaway**
Porting #49837 to master

### CODE REVIEW

**Review by dwoz — DISMISSED** [review]

**Comment by dwoz** [comment]
re-run full all

**Review by dhiltonp — DISMISSED** [review]

File: salt/states/file.py [hunk 1]
```diff
@@ -3956,9 +3956,19 @@ def recurse(name,
           - include_pat: E@hello      :: regexp matches 'otherhello',
                                          'hello01' ...
 
+        .. versionchanged:: Neon
```
Comment (DmitryKuzmenko) [inline salt/states/file.py:3959]: Sodium?

File: salt/states/file.py [hunk 2]
```diff
@@ -3973,6 +3983,15 @@ def recurse(name,
           - exclude_pat: E@(APPDATA)|(TEMPDATA) :: regexp matches APPDATA
                                                    or TEMPDATA for exclusion
 
+        .. versionchanged:: Neon
```
Comment (DmitryKuzmenko) [inline salt/states/file.py:3986]: Sodium?

**Review by DmitryKuzmenko — CHANGES_REQUESTED** [review]

**Comment by mchugh19** [comment]
@garethgreenaway are you able to fix up the doc strings? Or should I duplicate the PR to fix it up?

**Comment by garethgreenaway** [comment]
@mchugh19 Thanks for the ping, the versionchanged strings have been updated.

**Review by dwoz — APPROVED** [review]

**Pull request merged**